### DISCUSSION TITLE
Missing script: "build" after migration

### DISCUSSION THREAD

**Discussion by aldik1971** [ionic-team/capacitor#6160] [General]
Hello, I tried migrate from Cordova to Capacitor using this instructions on web page URL
https://capacitorjs.com/cordova

After migrations I tried to run command 

npm run build

but I got this error

npm ERR! Missing script: "build"
npm ERR! 
npm ERR! To see a list of scripts, run:
npm ERR!   npm run
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\aldik\AppData\Local\npm-cache\_logs\2022-12-12T11_13_29_759Z-debug-0.log (Log file is an attachment)

My package.json includes this section

"scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },

Can  anybody help me fix this issue - missing build scripts? I have tried multiple tutorials on the internet but without success.
(for example add to this section  "ionic:build": "ionic-app-scripts build",
    "ionic:serve": "ionic-app-scripts serve" )

Regards

Aldik

[log.txt](https://github.com/ionic-team/capacitor/files/10207494/log.txt)

**Comment by jcesarmobile** [comment]
This is what the link says

> # Most web app's
> npm run build

> # Ionic app
> ionic build

It means that most apps have a build script for building the app
If you don’t have a custom build script and you are using ionic, then run `ionic build` instead.

**Comment by aldik1971** [comment]
Hello, first, I'd like to apologize for the late response. 

I ran  "ionic build" and I got this message - [ERROR] Sorry! ionic build can only be run in an Ionic project directory.
As a next step I did "ionic init", fild app name and chose  as a project type - custom.

After that, I added to my package.json this code to scripts section:

  "test": "echo \"Error: no test specified\" && exit 1",
    "ionic:build": "ionic-app-scripts build",
    "ionic:serve": "ionic-app-scripts serve",
    "serve": "vue-cli-service serve",
    "build": "vue-cli-service build",
    "lint": "vue-cli-service lint"

As my next step I ran "npx cap sync" and "ionic build --verbose" 

I got this message: 

'ionic-app-scripts' is not recognized as an internal or external command,
operable program or batch file.
[ERROR] An error occurred while running subprocess npm.

        npm.cmd run ionic:build exited with exit code 1.

        Re-running this command with the --verbose flag may provide more information.
  ionic:utils-process onBeforeExit handler: 'process.exit' received +0ms
  ionic:utils-process onBeforeExit handler: running 1 functions +1ms
  ionic:utils-process error while killing process tree for 12452: Error: Command failed: taskkill /pid 12452 /T /F
  ionic:utils-process ERROR: The process "12452" not found.
  ionic:utils-process
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:412:12)
  ionic:utils-process     at ChildProcess.emit (node:events:513:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1091:16)
  ionic:utils-process     at Socket.<anonymous> (node:internal/child_process:449:11)
  ionic:utils-process     at Socket.emit (node:events:513:28)
  ionic:utils-process     at Pipe.<anonymous> (node:net:313:12) {
  ionic:utils-process   code: 128,
  ionic:utils-process   killed: false,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 12452 /T /F'
  ionic:utils-process } +209ms
  ionic:utils-process onBeforeExit handler: error from function: Error: Command failed: taskkill /pid 12452 /T /F
  ionic:utils-process ERROR: The process "12452" not found.
  ionic:utils-process
  ionic:utils-process     at ChildProcess.exithandler (node:child_process:412:12)
  ionic:utils-process     at ChildProcess.emit (node:events:513:28)
  ionic:utils-process     at maybeClose (node:internal/child_process:1091:16)
  ionic:utils-process     at Socket.<anonymous> (node:internal/child_process:449:11)
  ionic:utils-process     at Socket.emit (node:events:513:28)
  ionic:utils-process     at Pipe.<anonymous> (node:net:313:12) {
  ionic:utils-process   code: 128,
  ionic:utils-process   killed: false,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 12452 /T /F'
  ionic:utils-process } +2ms
  ionic:utils-process processExit: exiting (exit code: 1) +2ms


  After this message I tried install " npm install @ionic/app-scripts@latest --save-dev"  , but I got this message

  
npm ERR! gyp info using node-gyp@3.8.0
npm ERR! gyp info using node@18.12.1 | win32 | x64
npm ERR! gyp verb command rebuild []
npm ERR! gyp verb command clean []
npm ERR! gyp verb clean removing "build" directory
npm ERR! gyp verb command configure []
npm ERR! gyp verb check python checking for Python executable "python2" in the PATH
npm ERR! gyp verb `which` failed Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:13:12)
npm ERR! gyp verb `which` failed     at F (C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:68:19)
npm ERR! gyp verb `which` failed     at E (C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:80:29)
npm ERR! gyp verb `which` failed     at C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:89:16
npm ERR! gyp verb `which` failed     at C:\MyProject\node_modules\isexe\index.js:42:5
npm ERR! gyp verb `which` failed     at C:\MyProject\node_modules\isexe\windows.js:36:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:207:21)
npm ERR! gyp verb `which` failed  python2 Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:13:12)
npm ERR! gyp verb `which` failed     at F (C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:68:19)
npm ERR! gyp verb `which` failed     at E (C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:80:29)
npm ERR! gyp verb `which` failed     at C:\MyProject\node_modules\node-gyp\node_modules\which\which.js:89:16
npm ERR! gyp verb `which` failed     at C:\MyProject\node_modules\isexe\index.js:42:5
npm ERR! gyp verb `which` failed     at C:\MyProject\node_modules\isexe\windows.js:36:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:207:21) {
npm ERR! gyp verb `which` failed   code: 'ENOENT'
npm ERR! gyp verb `which` failed }
npm ERR! gyp verb check python checking for Python executable "python" in the PATH
npm ERR! gyp verb `which` succeeded python C:\Python311\python.EXE
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: Command failed: C:\Python311\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack   File "<string>", line 1
npm ERR! gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
npm ERR! gyp ERR! stack SyntaxError: Missing parentheses in call to 'print'. Did you mean print(...)?
npm ERR! gyp ERR! stack
npm ERR! gyp ERR! stack     at ChildProcess.exithandler (node:child_process:412:12)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
npm ERR! gyp ERR! stack     at Socket.<anonymous> (node:internal/child_process:449:11)
npm ERR! gyp ERR! stack     at Socket.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at Pipe.<anonymous> (node:net:313:12)
npm ERR! gyp ERR! System Windows_NT 10.0.19045
npm ERR! gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\MyProject\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd C:\MyProject\node_modules\node-sass
npm ERR! gyp ERR! node -v v18.12.1
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok
npm ERR! Build failed with error code: 1


Python is in .npmrc config file written as 
python "C:\Python27\python.exe"=true

and it is in system PATH too.

Thanks for your help me.

**Reply by jcesarmobile** [reply]
Sorry, I thought your app was an Ionic app. If it’s not an Ionic app then you don’t have to run ionic build.
How does your app folder structure looks like?

**Comment by aldik1971** [comment]
It is an Apache Cordova android application. Main folder has this structure

.vs                                                                  
.vscode 
-node_modules
-platforms
-plugins                                                              
-www                                                                  
.gitignore                                                           
.hintrc                                                              
.prettierrc                                                          
.project                                                             
conf.json                                                            
config.xml                                                           
jsconfig.json                                                        
package-lock.json                                                    
package.json                                                         



WWW folder includes

-components                                                           
-css                                                                  
-images                                                               
-js                                                                   
-js_main                                                              
-lib                                                                  
-resource                                                             
-ts                                                                   
.brackets.json                                                       
.prettierignore                                                      
Gruntfile.js                                                         
index.html                                                           
init-statements.sql                                                  
tsconfig.json                                                        
tslint.json    


After migration main folder structure includes                                                        

.vs                                                                  
.vscode                                                              
-android                                                              
-node_modules                                                         
-www                                                                  
.gitattributes                                                       
.gitignore                                                           
.hintrc                                                              
.npmrc                                                               
.prettierrc                                                          
.project                                                             
capacitor.config.json                                                
conf.json                                                            
ionic.config.json                                                    
jsconfig.json                                                        
package-lock.json                                                    
package.json                                                         


Thanks

**Reply by jcesarmobile** [reply]
Then if you already have a www folder that is not built, you don't need the build script.
But it's harder to use Capacitor if you are not using any javascript bundler as you can't use plugins as the documented usage.
